base/applications/charmap/map.c

```diff
@@ -206,6 +209,37 @@ MoveLargeCell(PMAP infoPtr)
 }
 
 
+static
+VOID
+GetPossibleCharacters(WCHAR* ch, INT codePageIdx)
```

Comment:
You should add to this function a new parameter that explicitly specifies the length of the buffer `ch` is pointing to.

Reply:
Good idea!